azkaban-web-server/src/test/java/azkaban/webapp/servlet/LoginAbstractAzkabanServletTestBase.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright 2020 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package azkaban.webapp.servlet;
+
+import azkaban.ServiceProvider;
+import azkaban.executor.ExecutableFlow;
+import azkaban.executor.ExecutorManagerAdapter;
+import azkaban.executor.ExecutorManagerException;
+import azkaban.flow.Flow;
+import azkaban.flowtrigger.FlowTriggerService;
+import azkaban.project.Project;
+import azkaban.project.ProjectManager;
+import azkaban.scheduler.ScheduleManager;
+import azkaban.server.session.Session;
+import azkaban.user.Permission;
+import azkaban.user.Permission.Type;
+import azkaban.user.Role;
+import azkaban.user.User;
+import azkaban.user.UserManager;
+import azkaban.utils.Props;
+import azkaban.webapp.AzkabanWebServer;
+import azkaban.webapp.WebMetrics;
+import com.google.common.collect.ImmutableMap;
+import com.google.inject.Injector;
+import java.io.IOException;
+import javax.servlet.ServletConfig;
+import org.codehaus.jackson.JsonNode;
+import org.codehaus.jackson.map.ObjectMapper;
+import org.junit.Before;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.mock.web.MockHttpServletResponse;
+
+public abstract class LoginAbstractAzkabanServletTestBase {
+
+  protected static final ObjectMapper MAPPER = new ObjectMapper();
+
+  @Mock
+  protected Injector injector;
+
+  @Mock
+  protected AzkabanWebServer azkabanWebServer;
+  @Mock
+  protected WebMetrics webMetrics;
+
+  @Mock
+  protected ProjectManager projectManager;
+  @Mock
+  protected UserManager userManager;
+  @Mock
+  protected ExecutorManagerAdapter executorManager;
+  @Mock
+  protected ScheduleManager scheduleManager;
+  @Mock
+  protected FlowTriggerService flowTriggerService;
+
+  @Mock
+  protected ServletConfig servletConfig;
+  @Mock
+  protected Session session;
+  protected AzkabanMockHttpServletRequest req;
+  protected AzkabanMockHttpServletResponse res;
+
+  @Captor
+  protected ArgumentCaptor<ExecutableFlow> exFlow;
```

Comment:
What's this used for?

Reply:
never mind I just found it.

Reply:
To catch value of such arg in mocked methods. The value is read in the test postAjaxExecuteFlowWithSlaSettings.